### DISCUSSION TITLE
How is window transparency managed and how can I tone it down/remove it?

### DISCUSSION THREAD

**Discussion by vincentcomfy** [prasanthrangan/hyprdots#1702] [Q&A] [answered]
I've gone through all 18 pages of discussions and the only thing that I've found that's close is someone asking how to modify the opacity of the TOR browser. I tried the comments in there and they weren't recognized in my terminal, so I can only assume it's for something different and with different dependencies. I also checked the wiki and front page of this repo but couldn't find anything (the wiki was pretty much empty, not sure if it just failed to load?)

I also asked gemini and looked through the arch wiki, but it seems like the transparency is being managed by a theming manager/app and I have no experience with any that are installed.

I'd like to basically remove transparency from Firefox and certain other apps to make the desktop background not interfere with the content I'm viewing. I don't mind it for the terminal and some other apps, but for Firefox it's just too much for me. Can someone please advis

**Comment by kRHYME7** [comment, marked as answer]
in the file `~/.config/hypr/windowsrules.conf`

**Reply by vincentcomfy** [reply]
Thank you very much!

**Comment by Shubh-Goyal-07** [comment]
I changed the opacity values everywhere to 1 but my windows are still translucent. Can you help?

**Reply by kRHYME7** [reply]
use `override` keyword. 

See the note in this section https://wiki.hyprland.org/Configuring/Window-Rules/#syntax

**Reply by Shubh-Goyal-07** [reply]
```
  8 $&=override
  9 
 10 windowrulev2 = opacity 1 $& 1 $& 1,class:^(firefox)$
 11 windowrulev2 = opacity 1 $& 1 $& 1,class:^(Brave-browser)$
```

this is the rule in my .config/hypr/windowrules.conf

**Reply by kRHYME7** [reply]
which file again? like the absolute path please.

**Reply by Shubh-Goyal-07** [reply]
/home/username/.config/hypr/windowrules.conf

**Reply by kRHYME7** [reply]
try adding this in userprefs.conf

`
windowrulev2 = opacity 1 override 1 override 1 override,class:^(.*)$
`

**Reply by Shubh-Goyal-07** [reply]
Didn't work ;-;

**Reply by ffyydos** [reply]
I believe the class should be "brave-browser" instead of "Brave-browser", that did the trick for me
